src/Illuminate/Database/Eloquent/Relations/Concerns/InteractsWithPivotTable.php

```diff
@@ -348,12 +348,14 @@ public function detach($ids = null, $touch = true)
         // If associated IDs were passed to the method we will only delete those
         // associations, otherwise all of the association ties will be broken.
         // We'll return the numbers of affected rows when we do the deletes.
-        if (! is_null($ids = $this->parseIds($ids))) {
-            if (count((array) $ids) === 0) {
+        if (! is_null($ids)) {
+            $ids = $this->parseIds($ids);
```

Comment:
@laurencei Yep that's exactly it however IMO that seems more like a bug than a feature so the change shouldn't matter... 😄